internal/Picker/Picker.js

```diff
@@ -290,39 +319,81 @@ const PickerBase = kind({
 			handleDecrement,
 			handleFlick,
 			handleIncrement,
-			handleSecondaryDecrement,
-			handleSecondaryIncrement,
 			incrementAriaLabel: incAriaLabel,
+			index,
 			min,
 			max,
+			noAnimation,
+			orientation,
+			reverseTransition,
 			skin,
 			step,
 			value,
 			valueId,
 			width,
 			...rest
 		} = props;
+
 		const currentValue = Array.isArray(values) ? values[value] : value;
-		const decrementValue = clamp(min, max, Array.isArray(values) ? values[value - step] : value - step);
-		const incrementValue = clamp(min, max, Array.isArray(values) ? values[value + step] : value + step);
-		const secondaryDecrementValue = clamp(min, max, Array.isArray(values) ? values[value - (2 * step)] : value - (2 * step));
-		const secondaryIncrementValue = clamp(min, max, Array.isArray(values) ? values[value + (2 * step)] : value + (2 * step));
 		const isFirst = value <= min;
 		const isLast = value >= max;
 		const isSecond = value <= min + step;
 		const isPenultimate = value >= max - step;
 		const decrementAriaLabel = `${currentValueText} ${decAriaLabel}`;
 		const incrementAriaLabel = `${currentValueText} ${incAriaLabel}`;
 
+		const decrementValue = () => {
+			if (isFirst) {
+				return '';
+			} else if (Array.isArray(values)) {
+				return values;
+			} else {
+				return (<PickerItem key={clamp(min, max, value - step)} style={{direction: 'ltr'}}>{clamp(min, max, value - step)}</PickerItem>);
+			}
+		};
+
+		const incrementValue = () => {
+			if (isLast) {
+				return '';
+			} else if (Array.isArray(values)) {
+				return values;
+			} else {
+				return (<PickerItem key={clamp(min, max, value + step)} style={{direction: 'ltr'}}>{clamp(min, max, value + step)}</PickerItem>);
+			}
+		};
+
+		const secondaryDecrementValue = () => {
+			if (isSecond) {
+				return '';
+			} else if (Array.isArray(values)) {
+				return values;
+			} else {
+				return (<PickerItem key={clamp(min, max, value - (2 * step))} style={{direction: 'ltr'}}>{clamp(min, max, value - (2 * step))}</PickerItem>);
+			}
+		};
+
+		const secondaryIncrementValue = () => {
+			if (isPenultimate) {
+				return '';
+			} else if (Array.isArray(values)) {
+				return values;
+			} else {
+				return (<PickerItem key={clamp(min, max, value + (2 * step))} style={{direction: 'ltr'}}>{clamp(min, max, value + (2 * step))}</PickerItem>);
+			}
+		};
+
 		let sizingPlaceholder = null;
 		if (typeof width === 'number' && width > 0) {
 			sizingPlaceholder = <div aria-hidden className={css.sizingPlaceholder}>{'0'.repeat(width)}</div>;
 		}
 
-		delete rest.accessibilityHint;
 		delete rest['aria-valuetext'];
+		delete rest.accessibilityHint;
 		delete rest.orientation;
 
+		const horizontal = orientation === 'horizontal';
+		let arranger = horizontal ? SlideLeftArranger : SlideTopArranger;
```

Comment:
let's use const and move this line and the previous one above the "delete rest."  lines